src/Avalonia.Controls/Control.cs

```diff
@@ -372,23 +372,19 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
         /// </summary>
         public virtual void InvalidateMirrorTransform()
         {
-            var flowDirection = this.FlowDirection;
+            var flowDirection = FlowDirection;
             var parentFlowDirection = FlowDirection.LeftToRight;
 
             bool bypassFlowDirectionPolicies = BypassFlowDirectionPolicies;
             bool parentBypassFlowDirectionPolicies = false;
 
-            var parent = this.FindAncestorOfType<Control>();
+            var parent = ((IVisual)this).VisualParent as Control;
```

Comment:
What happens if the parent is not `Control`, the chain will break, isn't?

Reply:
I can't think of a scenario where that happens. Let's do some profiling. Ideally we move FlowDirection to Visual and let the renderer do the mirroring.

Reply:
If FlowDirection is moved to Visual it should work.

If you are planning to delete HasMirrorTransform, it seems to me that this is not a good idea, there should be an indication whether MirrorTransform apply on Control.